analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -212,6 +215,61 @@ public void Verify_UnchangedFiles(string unchangedFileName, bool expectedProtobu
                         VerifyReferences(orderedSymbols[1].Reference, 9, 44, 41);           // LocalMethod
                     });
 
+        [DataTestMethod]
+        [DataRow(ProjectType.Product)]
+        [DataRow(ProjectType.Test)]
+        public void Verify_PrimaryConstructor_PreciseLocation_CSharp12(ProjectType projectType) =>
+            Verify("PrimaryConstructor.cs", projectType, references =>
+            {
+                references.Select(x => x.Declaration.StartLine).Should().BeEquivalentTo(new[] { 1, 3, 6, 6, 8, 8, 10, 11, 12, 14, 17, 17, 19, 20, 21, 21, 23, 23, 25 });
+
+                var primaryCtorParameterDeclaration = references.Single(x => x.Declaration.StartLine == 8 && x.Declaration.StartOffset == 19); // b1, primary ctor
```

Comment:
`primaryCtorParameterDeclaration` can be shortened to `primaryCtorParameter` without losing information, given the context.